File: algorithms/binarySearch.js
function binarySearch(nums, target) {
  let start = 0;
  let end = nums.length - 1;
  let mid = Math.floor((start + end) / 2);

  while (start <= end) {
    if (target === nums[mid]) return mid;
    else if (target > nums[mid]) start = mid + 1;
    else end = mid - 1;
    mid = Math.floor((start + end) / 2);
  }
  return -1;
}

File: algorithms/binary-search.py

def binary_search(nums, target): # Time: O(log(n)), Space: O(1)
    left, right = 0, len(nums) - 1
    while left <= right:
        mid = (left+right) // 2
        if target == nums[mid]: return mid
        elif target > nums[mid]: left = mid + 1
        else: right = mid - 1
    return -1